onnxruntime/core/graph/ep_api_types.cc

```diff
@@ -91,6 +91,22 @@ Status EpNode::Create(const Node& node, const EpGraph* ep_graph,
   ConvertNodeArgsToValueInfos(ep_graph, value_infos_map, node_inputs, ep_node_inputs);
   ConvertNodeArgsToValueInfos(ep_graph, value_infos_map, node_outputs, ep_node_outputs);
 
+  const auto& node_attrs = node.GetAttributes();
+  std::unordered_map<std::string, std::unique_ptr<ONNX_NAMESPACE::AttributeProto>> ep_node_attributes_map;
+  std::vector<OrtOpAttr*> ep_node_attributes;
+
+  if (node_attrs.size() > 0) {
+    ep_node_attributes.resize(node_attrs.size(), nullptr);
```

Comment:
nit: could use `ep_node_attributes.reserve(node_attrs.size())` combined with `ep_node_attributes.push_back()`. This removes need for `attr_idx` and gets rid of an unnecessary initialization to `nullptr`.